app/models/concerns/arclight/catalog.rb

```diff
@@ -4,6 +4,18 @@ module Arclight
   ##
   # Arclight specific methods for the Catalog
   module Catalog
+    extend ActiveSupport::Concern
+
+    included do
+      before_action only: :index do
+        if params.dig(:f, :collection_sim) && params[:within_collection] == 'false'
+          params[:f].delete(:collection_sim)
+          params.delete(:within_collection)
+          redirect_to url_for(params.to_unsafe_h)
```

Comment:
Is there a better way to do this?

Reply:
What if we gave the dropdown a little more smarts (probably better than this, but..)?

```
...
  <select
    ...
    name="f[collection_sim][]"
  >
    <option value="">
      ...
    </option>
    <option
      value="<%= Array(params[:f][:collection_sim]).first || @document && @document.collection_name %>">
     ...
    </option>
  </select>
...
```

Reply:
Oh, and always exclude it from the hidden fields:

```erb
    <%= render_hash_as_hidden_fields(search_state.params_for_search.except(:q, :search_field, :qt, :page, :utf8, :f).merge(f: search_state.params_for_search[:f].except(:collection_sim))) %>
```

Reply:
And, I guess, we still need something to prune empty values from the params.. ugh.

Reply:
Yeah, it appears that we would, and also need to redirect unless we want to carry around an empty collection_sim facet value in the params (and just ignore it).  I'm indifferent about the approach.  One one hand, doing this doesn't add an additional parameter to the URL but on the other seems to be a little less clean wrt to the facet parameters hanging around and in-view parameter manipulation.

I'm happy to stamp out this approach if it is deemed preferable.

Reply:
@cbeer let me know if what I've done is in the spirit of what you were thinking (and if there is any potential issues w/ the ignored `f` param that I was trying to avoid w/ my redirect).